Fix spi env

mirceacaprioru opened the pull request.

Commit:
configs: zynq-common: Guard against M2K platform

Some environment configurations are only valid for Pluto. These patch does
a check for the board model and guard against calling configurations for
M2k.

Signed-off-by: Mircea Caprioru <mircea.caprioru@analog.com>

Commit:
configs: zynq-common: Fix mode 2r2t

For AD9364 there is only mode 1r1t available but this condition was
enforced for all compatible devices. This patch fixes the mode option for
AD9361 and AD9363.

Fixes: e1eb71f0290d ("configs: zynq-common: Fix SPI env update")
Signed-off-by: Mircea Caprioru <mircea.caprioru@analog.com>

mhennerich commented:
m2k and Pluto Rev.C looks clean now!
However Pluto Rev.A and Rev.B still shows errors 

**libfdt fdt_path_offset() returned FDT_ERR_NOTFOUND
libfdt fdt_path_offset() returned FDT_ERR_NOTFOUND
libfdt fdt_delprop():  FDT_ERR_NOTFOUND**


```
U-Boot PlutoSDR v0.20-PlutoSDR-00059-g4ae641acba (Dec 18 2020 - 08:24:24 +0100)

I2C:   ready
DRAM:  ECC disabled 512 MiB
SF: Detected N25Q256A with page size 256 Bytes, erase size 4 KiB, total 32 MiB
In:    serial@e0001000
Out:   serial@e0001000
Err:   serial@e0001000
Model: Zynq Pluto SDR Board
Hit any key to stop autoboot:  0 
Copying Linux from QSPI flash to RAM...
SF: Detected N25Q256A with page size 256 Bytes, erase size 4 KiB, total 32 MiB
device 0 offset 0xff000, size 0x1000
SF: 4096 bytes @ 0xff000 Read: OK
## Error: bad CRC, import failed
device 0 offset 0x200000, size 0x900000
SF: 9437184 bytes @ 0x200000 Read: OK

## Checking Image at 02080000 ...
   FIT image found
   FIT description: Configuration to load fpga before Kernel
    Image 0 (fdt@1)
     Description:  zynq-pluto-sdr
     Type:         Flat Device Tree
     Compression:  uncompressed
     Data Start:   0x02080100
     Data Size:    16512 Bytes = 16.1 KiB
     Architecture: ARM
    Image 1 (fdt@2)
     Description:  zynq-pluto-sdr-revb
     Type:         Flat Device Tree
     Compression:  uncompressed
     Data Start:   0x020841f4
     Data Size:    16588 Bytes = 16.2 KiB
     Architecture: ARM
    Image 2 (fdt@3)
     Description:  zynq-pluto-sdr-revc
     Type:         Flat Device Tree
     Compression:  uncompressed
     Data Start:   0x02088334
     Data Size:    17056 Bytes = 16.7 KiB
     Architecture: ARM
    Image 3 (fpga@1)
     Description:  FPGA
     Type:         FPGA Image
     Compression:  uncompressed
     Data Start:   0x0208c63c
     Data Size:    964480 Bytes = 941.9 KiB
     Load Address: 0x0f000000
     Hash algo:    md5
     Hash value:   45534aec1cae2d9628a94b2a0d7a2cfc
    Image 4 (linux_kernel@1)
     Description:  Linux
     Type:         Kernel Image
     Compression:  uncompressed
     Data Start:   0x02177e78
     Data Size:    3912880 Bytes = 3.7 MiB
     Architecture: ARM
     OS:           Linux
     Load Address: 0x00008000
     Entry Point:  0x00008000
     Hash algo:    md5
     Hash value:   a26fdff2cca953c5902c13502ceb7eda
    Image 5 (ramdisk@1)
     Description:  Ramdisk
     Type:         RAMDisk Image
     Compression:  gzip compressed
     Data Start:   0x02533404
     Data Size:    5694596 Bytes = 5.4 MiB
     Architecture: ARM
     OS:           Linux
     Load Address: unavailable
     Entry Point:  unavailable
     Hash algo:    md5
     Hash value:   6769994294cac50b3756c3edba6cad70
    Default Configuration: 'config@0'
    Configuration 0 (config@0)
     Description:  Linux with fpga RevA
     Kernel:       linux_kernel@1
     Init Ramdisk: ramdisk@1
     FDT:          fdt@1
     FPGA:         fpga@1
    Configuration 1 (config@1)
     Description:  Linux with fpga RevB
     Kernel:       linux_kernel@1
     Init Ramdisk: ramdisk@1
     FDT:          fdt@2
     FPGA:         fpga@1
    Configuration 2 (config@2)
     Description:  Linux with fpga RevB
     Kernel:       linux_kernel@1
     Init Ramdisk: ramdisk@1
     FDT:          fdt@2
     FPGA:         fpga@1
    Configuration 3 (config@3)
     Description:  Linux with fpga RevB
     Kernel:       linux_kernel@1
     Init Ramdisk: ramdisk@1
     FDT:          fdt@2
     FPGA:         fpga@1
    Configuration 4 (config@4)
     Description:  Linux with fpga RevB
     Kernel:       linux_kernel@1
     Init Ramdisk: ramdisk@1
     FDT:          fdt@2
     FPGA:         fpga@1
    Configuration 5 (config@5)
     Description:  Linux with fpga RevB
     Kernel:       linux_kernel@1
     Init Ramdisk: ramdisk@1
     FDT:          fdt@2
     FPGA:         fpga@1
    Configuration 6 (config@6)
     Description:  Linux with fpga RevB
     Kernel:       linux_kernel@1
     Init Ramdisk: ramdisk@1
     FDT:          fdt@2
     FPGA:         fpga@1
    Configuration 7 (config@7)
     Description:  Linux with fpga RevB
     Kernel:       linux_kernel@1
     Init Ramdisk: ramdisk@1
     FDT:          fdt@2
     FPGA:         fpga@1
    Configuration 8 (config@8)
     Description:  Linux with fpga RevC
     Kernel:       linux_kernel@1
     Init Ramdisk: ramdisk@1
     FDT:          fdt@3
     FPGA:         fpga@1
    Configuration 9 (config@9)
     Description:  Linux with fpga RevB
     Kernel:       linux_kernel@1
     Init Ramdisk: ramdisk@1
     FDT:          fdt@2
     FPGA:         fpga@1
    Configuration 10 (config@10)
     Description:  Linux with fpga RevB
     Kernel:       linux_kernel@1
     Init Ramdisk: ramdisk@1
     FDT:          fdt@2
     FPGA:         fpga@1
## Checking hash(es) for FIT Image at 02080000 ...
   Hash(es) for Image 0 (fdt@1): 
   Hash(es) for Image 1 (fdt@2): 
   Hash(es) for Image 2 (fdt@3): 
   Hash(es) for Image 3 (fpga@1): md5+ 
   Hash(es) for Image 4 (linux_kernel@1): md5+ 
   Hash(es) for Image 5 (ramdisk@1): md5+ 
libfdt fdt_path_offset() returned FDT_ERR_NOTFOUND
libfdt fdt_path_offset() returned FDT_ERR_NOTFOUND
libfdt fdt_delprop():  FDT_ERR_NOTFOUND
fdt - flattened device tree utility commands

Usage:
fdt addr [-c]  <addr> [<length>]   - Set the [control] fdt location to <addr>
fdt move   <fdt> <newaddr> <length> - Copy the fdt to <addr> and make it active
fdt resize                          - Resize fdt to size + padding to 4k addr
fdt print  <path> [<prop>]          - Recursive print starting at <path>
fdt list   <path> [<prop>]          - Print one level starting at <path>
fdt get value <var> <path> <prop>   - Get <property> and store in <var>
fdt get name <var> <path> <index>   - Get name of node <index> and store in <var>
fdt get addr <var> <path> <prop>    - Get start address of <property> and store in <var>
fdt get size <var> <path> [<prop>]  - Get size of [<property>] or num nodes and store in <var>
fdt set    <path> <prop> [<val>]    - Set <property> [to <val>]
fdt mknode <path> <node>            - Create a new node after <path>
fdt rm     <path> [<prop>]          - Delete the node or <property>
fdt header                          - Display header info
fdt bootcpu <id>                    - Set boot cpuid
fdt memory <addr> <size>            - Add/Update memory node
fdt rsvmem print                    - Show current mem reserves
fdt rsvmem add <addr> <size>        - Add a mem reserve
fdt rsvmem delete <index>           - Delete a mem reserves
fdt chosen [<start> <end>]          - Add/update the /chosen branch in the tree
                                        <start>/<end> - initrd start/end addr
fdt checksign [<addr>]              - check FIT signature
                                        <start> - addr of key blob
                                                  default gd->fdt_blob
NOTE: Dereference aliases by omitting the leading '/', e.g. fdt print ethernet0.
## Loading kernel from FIT Image at 02080000 ...
   Using 'config@0' configuration
   Verifying Hash Integrity ... OK
   Trying 'linux_kernel@1' kernel subimage
     Description:  Linux
     Type:         Kernel Image
     Compression:  uncompressed
     Data Start:   0x02177e78
     Data Size:    3912880 Bytes = 3.7 MiB
     Architecture: ARM
     OS:           Linux
     Load Address: 0x00008000
     Entry Point:  0x00008000
     Hash algo:    md5
     Hash value:   a26fdff2cca953c5902c13502ceb7eda
   Verifying Hash Integrity ... md5+ OK
## Loading ramdisk from FIT Image at 02080000 ...
   Using 'config@0' configuration
   Trying 'ramdisk@1' ramdisk subimage
     Description:  Ramdisk
     Type:         RAMDisk Image
     Compression:  gzip compressed
     Data Start:   0x02533404
     Data Size:    5694596 Bytes = 5.4 MiB
     Architecture: ARM
     OS:           Linux
     Load Address: unavailable
     Entry Point:  unavailable
     Hash algo:    md5
     Hash value:   6769994294cac50b3756c3edba6cad70
   Verifying Hash Integrity ... md5+ OK
## Loading fdt from FIT Image at 02080000 ...
   Using 'config@0' configuration
   Trying 'fdt@1' fdt subimage
     Description:  zynq-pluto-sdr
     Type:         Flat Device Tree
     Compression:  uncompressed
     Data Start:   0x02080100
     Data Size:    16512 Bytes = 16.1 KiB
     Architecture: ARM
   Verifying Hash Integrity ... OK
   Booting using the fdt blob at 0x2080100
## Loading fpga from FIT Image at 02080000 ...
   Trying 'fpga@1' fpga subimage
     Description:  FPGA
     Type:         FPGA Image
     Compression:  uncompressed
     Data Start:   0x0208c63c
     Data Size:    964480 Bytes = 941.9 KiB
     Load Address: 0x0f000000
     Hash algo:    md5
     Hash value:   45534aec1cae2d9628a94b2a0d7a2cfc
   Verifying Hash Integrity ... md5+ OK
   Loading fpga from 0x0208c63c to 0x0f000000
  design filename = "system_top;UserID=0XFFFFFFFF;COMPRESS=TRUE;Version=2019.1"
  part number = "7z010clg225"
  date = "2020/10/28"
  time = "09:55:53"
  bytes in bitstream = 964360
zynq_align_dma_buffer: Align buffer at f000078 to f000000(swap 1)
   Programming full bitstream... OK
   Loading Kernel Image ... OK
   Loading Ramdisk to 1e5ba000, end 1eb28484 ... OK
   Loading Device Tree to 1e5b2000, end 1e5b907f ... OK

Starting kernel ...
```

Commit:
configs: zynq-common: Fix rev B fdt errors

For Pluto Rev B there are no 2r2t options so these do not have to be
removed. The same goes for refclk_source which is not available for Rev. B.

Signed-off-by: Mircea Caprioru <mircea.caprioru@analog.com>

mirceacaprioru commented:
changelog v2
- fix fdt errors for rev. B

Commit:
configs: zynq-common: Use double quotes for strings

It is safer to use quotes for variables containing string values in bash.
Also the shell used in u-boot is limited with string operations and only
accomodats the use of a minimal test command.

Signed-off-by: Mircea Caprioru <mircea.caprioru@analog.com>

Commit:
config: zynq-common: Fix refclk setting

Reorder the operations done for refclk setting. Check first that the
setting is valid and then apply the changes.

Signed-off-by: Mircea Caprioru <mircea.caprioru@analog.com>

Commit:
config: zynq-common: Fix attr_val attr_name

Reorder the operations for attr_val and check that the values are valid
before applying the changes.

Signed-off-by: Mircea Caprioru <mircea.caprioru@analog.com>

Commit:
config: zynq-common: Remove compatible env setting

This patch removes the extra compatible setting and only leaves the old
method for changing the IIO devicei, attr_name and attr_val.

The user will use the combination of these two commands:

fw_setenv attr_name compatible
fw_setenv attr_val ad9364

This will ensure that we don't change anything to the documentation that is
already in place for Pluto.

Signed-off-by: Mircea Caprioru <mircea.caprioru@analog.com>

mirceacaprioru commented:
Changelog v3
- removed the compatible setting to return just to the old method of attr_name and attr_val
- changed the logic for refclk_source and compatible to remove unneeded operations

Tested on Pluto rev. B/C

mhennerich commented on include/configs/zynq-common.h:
```diff
@@ -284,6 +285,9 @@
                 "fi; " \
 		"if test \"${refclk_source}\" = \"external\" && test \"${model}\" = \"Analog Devices PlutoSDR Rev.C (Z7010/AD9363)\" ; then " \
 			"fdt rm /amba/gpio@e000a000/clock_internal_en; " \
+			"if test -n \"${ad936x_ext_refclk_override}\"; then " \
```
Can we make ad936x_ext_refclk_override also available for Rev.A and B devices?
I see no reason why not. The problem is that ad936x_ext_refclk get's always overwritten by the extra-env

mirceacaprioru replied:
Yes I forgot about that overwrite. I can move it outside.

mirceacaprioru commented:
changelog v4

- make ad936x_ext_refclk_override also available for Rev.A and B devices

mirceacaprioru commented:
will merge later today if there are no more comments.
